ui/kotlinx-coroutines-android/test/HandlerDispatcherTest.kt

```diff
@@ -9,173 +9,131 @@ import kotlinx.coroutines.*
 import org.junit.Test
 import org.junit.runner.*
 import org.robolectric.*
-import org.robolectric.Shadows.*
 import org.robolectric.annotation.*
 import org.robolectric.shadows.*
-import org.robolectric.util.*
+import java.util.concurrent.*
 import kotlin.test.*
 
 @RunWith(RobolectricTestRunner::class)
 @Config(manifest = Config.NONE, sdk = [28])
 class HandlerDispatcherTest : TestBase() {
-
-    /**
-     * Because [Dispatchers.Main] is a singleton, we cannot vary its initialization behavior. As a
-     * result we only test its behavior on the newest API level and assert that it uses async
-     * messages. We rely on the other tests to exercise the variance of the mechanism that the main
-     * dispatcher uses to ensure it has correct behavior on all API levels.
-     */
     @Test
-    fun mainIsAsync() = runTest {
-        ReflectionHelpers.setStaticField(Build.VERSION::class.java, "SDK_INT", 28)
-
-        val mainLooper = shadowOf(Looper.getMainLooper())
-        mainLooper.pause()
-        val mainMessageQueue = shadowOf(Looper.getMainLooper().queue)
-
-        val job = launch(Dispatchers.Main) {
+    fun testImmediateDispatcherYield() = runBlocking(Dispatchers.Main) {
+        expect(1)
+        // launch in the immediate dispatcher
+        launch(Dispatchers.Main.immediate) {
             expect(2)
+            yield()
+            expect(4)
         }
-
```

Comment:
Life hack: `git diff --patience` provides a sensible diff instead of this broken mess with methods broken in half. Alas, doesn't seem like there's a way to make Github use another diffing algorithm.